boot/zephyr/include/sysflash/sysflash.h

```diff
@@ -69,7 +71,19 @@ extern uint32_t _image_1_primary_slot_id[];
 #if !defined(CONFIG_BOOT_SWAP_USING_MOVE)
 #define FLASH_AREA_IMAGE_SCRATCH    FLASH_AREA_ID(image_scratch)
 #endif
-#endif /* USE_PARTITION_MANAGER */
 
+#else /* CONFIG_SINGLE_IMAGE_DFU */
+
+#define FLASH_AREA_IMAGE_PRIMARY(x)	FLASH_AREA_ID(image_0)
+#define FLASH_AREA_IMAGE_SECONDARY(x)	FLASH_AREA_ID(image_0)
+/* NOTE: Scratch parition is not used by single image DFU but some of
+ * functions in common files reference it, so the definitions has been
+ * provided to allow compilation of common units.
+ */
+#define FLASH_AREA_IMAGE_SCRATCH	0
+
+#endif /* CONFIG_SINGLE_IMAGE_DFU */
+
+#endif /* USE_PARTITION_MANAGER */
```

Comment:
I think that https://github.com/nrfconnect/sdk-mcuboot/pull/99/commits/36ea233dc699a85d6a1b17f3994648aec775345c commit, from https://github.com/nrfconnect/sdk-mcuboot/pull/99/commits, may be needed to make it work. I have compared my PR branch with the upmerge branch and the commit is the difference.

Reply:
Thanks, I applied it now.